api/src/main/java/com/community/api/endpoint/avisoft/CartEndPoint.java

```diff
@@ -0,0 +1,111 @@
+package com.community.api.endpoint.avisoft;
+import com.broadleafcommerce.rest.api.endpoint.BaseEndpoint;
+import com.broadleafcommerce.rest.api.endpoint.order.CartEndpoint;
+import org.broadleafcommerce.core.order.domain.Order;
+import org.broadleafcommerce.core.order.service.OrderService;
+
+import org.broadleafcommerce.profile.core.domain.Customer;
+import org.broadleafcommerce.profile.core.service.CustomerService;
+import org.broadleafcommerce.profile.web.core.CustomerState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+@RestController
+@RequestMapping(value = "/cart-custom",
+        produces = { MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE }
+)
+public class CartEndPoint extends BaseEndpoint {
+    private static final Logger logger = LoggerFactory.getLogger(CartEndpoint.class);
+    @Autowired
+    private CustomerService customerService;
+    @Autowired
+    OrderService orderService;
+
+    @RequestMapping(value = "getCart/{customerId}", method = RequestMethod.GET)
+    public ResponseEntity<String> retrieveProductById(@PathVariable Long customerId) {
+        logger.debug("Retrieving customer by ID: {}", customerId);
+        Customer customer = customerService.readCustomerById(customerId);//finding the customer to get cart associated with it
```

Comment:
@kshtjvats remove this line as in try block you are checking null condition then it doesn't make sense

Reply:
also      
   if (customer == null) {
            customer = customerService.createNewCustomer();

            customer.setAnonymous(true);

            CustomerState.setCustomer(customer);

            customerService.saveCustomer(customer);

        }
        this code should be just below to readcustomerid function where u are getting data of customer